cli/internal/cmd/iamdestroy.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/cli/internal/cloudcmd"
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"github.com/edgelesssys/constellation/v2/internal/file"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func runDestroyIAMUser(cmd *cobra.Command, _args []string) error {
+	spinner := newSpinner(cmd.ErrOrStderr())
+	destroyer := cloudcmd.NewIAMDestroyer(cmd.Context())
+	fsHandler := file.NewHandler(afero.NewOsFs())
+
+	return destroyIAMUser(cmd, spinner, destroyer, fsHandler)
+}
+
+func destroyIAMUser(cmd *cobra.Command, spinner spinnerInterf, destroyer iamDestroyer, fsHandler file.Handler) error {
+	yes, err := cmd.Flags().GetBool("yes")
+	if err != nil {
+		return err
+	}
+
+	if !yes {
+		// Confirmation
+		ok, err := askToConfirm(cmd, "Do you really want to destroy your IAM user?")
+		if err != nil {
+			return err
+		}
+		if !ok {
+			fmt.Println("The destruction of the IAM user was aborted")
+			return nil
+		}
+	}
+
+	if proceed, err := deleteGCPServiceAccountKeyFile(cmd, destroyer, fsHandler); err != nil && !proceed {
+		return err
+	}
```

Comment:
If `proceed == false` and `err == nil`, the function will not return here.

Maybe the confirm/abort flow from the iam-create command would work here?
```suggestion
 	proceed, err := deleteGCPServiceAccountKeyFile(cmd, destroyer, fsHandler)
	if err != nil {
		return err
	}
	if !proceed {
		cmd.Println("Destruction was aborted.")
		return nil
	}
```

Please note that this will probably require some changes in the `deleteGCPServiceAccountKeyFile` function.
Mainly the call to `fsHandler.Stat()` has to check if the received error was of type `os.ErrNotExists`, and if so, return `false, nil`

Reply:
Returning `true, nil` makes more sense to me since a user can be destroyed if the file doesn't exist. This means that the function could be used in Azure or AWS environments without problems. I've added an additional check to catch errors beside `os.ErrNotExist` and added  a check in `destroyIAMUser`, do you think that this works as an alternative?